.github/workflows/publish-packages.yml

```
@@ -119,20 +125,23 @@ jobs:
       GH_TOKEN: ${{ secrets[needs.automation.outputs.token-secret-name] }}
 
     steps:
-      - name: check out code
+      - name: Check out code
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
-          token: ${{ secrets[needs.automation.outputs.token-secret-name] }}
+          token: ${{ env.GH_TOKEN }}
 
       - name: Download Artifacts
+        env:
+          ARTIFACT_ID: ${{ needs.build-pack-publish.outputs.artifact-id }}
+          ARTIFACT_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
           curl \
             -H "Accept: application/vnd.github+json" \
-            -H "Authorization: token ${{ github.token }}" \
+            -H "Authorization: token ${ARTIFACT_TOKEN}" \
```

Similar for following lines.

```suggestion
            -H "Authorization: token ${env:ARTIFACT_TOKEN}" \
```

---

I overlooked it.